packages/autoloader/src/AutoloadGenerator.php

```diff
@@ -0,0 +1,348 @@
+<?php
+/**
+ * Autoloader Generator.
+ *
+ * @package Automattic\Jetpack\Autoloader
+ */
+
+// phpcs:disable PHPCompatibility.Keywords.NewKeywords.t_useFound
+// phpcs:disable PHPCompatibility.LanguageConstructs.NewLanguageConstructs.t_ns_separatorFound
+// phpcs:disable PHPCompatibility.FunctionDeclarations.NewClosure.Found
+// phpcs:disable PHPCompatibility.Keywords.NewKeywords.t_namespaceFound
+// phpcs:disable PHPCompatibility.Keywords.NewKeywords.t_dirFound
+// phpcs:disable WordPress.Files.FileName.InvalidClassFileName
+// phpcs:disable WordPress.Files.FileName.NotHyphenatedLowercase
+// phpcs:disable WordPress.Files.FileName.InvalidClassFileName
+// phpcs:disable WordPress.PHP.DevelopmentFunctions.error_log_var_export
+// phpcs:disable WordPress.WP.AlternativeFunctions.file_system_read_file_put_contents
+// phpcs:disable WordPress.WP.AlternativeFunctions.file_system_read_fopen
+// phpcs:disable WordPress.WP.AlternativeFunctions.file_system_read_fwrite
+// phpcs:disable WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+// phpcs:disable WordPress.NamingConventions.ValidVariableName.InterpolatedVariableNotSnakeCase
+// phpcs:disable WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+// phpcs:disable WordPress.NamingConventions.ValidVariableName.PropertyNotSnakeCase
+
+namespace Automattic\Jetpack\Autoloader;
+
+use Composer\Autoload\AutoloadGenerator as BaseGenerator;
+use Composer\Autoload\ClassMapGenerator;
+use Composer\Config;
+use Composer\Installer\InstallationManager;
+use Composer\IO\IOInterface;
+use Composer\Package\PackageInterface;
+use Composer\Repository\InstalledRepositoryInterface;
+use Composer\Util\Filesystem;
+
+/**
+ * Class AutoloadGenerator.
+ */
+class AutoloadGenerator extends BaseGenerator {
+
+	/**
+	 * Whether or not generated autoloader considers the class map authoritative.
+	 *
+	 * @var bool Whether or not generated autoloader considers the class map authoritative.
+	 */
+	private $classMapAuthoritative = false;
+
+	/**
+	 * Instantiate an AutoloadGenerator object.
+	 *
+	 * @param IOInterface $io IO object.
+	 */
+	public function __construct( $io ) {
```

Comment:
Should we be stricter here?

```suggestion
	public function __construct( IOInterface $io = null ) {
```